src/components/MagicCodeInput.tsx

```diff
@@ -226,110 +221,108 @@ function MagicCodeInput(props) {
      * the focused input on the next empty one, if exists.
      * It handles both fast typing and only one digit at a time
      * in a specific position.
-     *
-     * @param {String} value
      */
-    const onChangeText = (value) => {
-        if (_.isUndefined(value) || _.isEmpty(value) || !ValidationUtils.isNumeric(value)) {
+    const onChangeText = (textValue?: string) => {
+        if (!textValue?.length || !ValidationUtils.isNumeric(textValue)) {
             return;
         }
 
         // Checks if one new character was added, or if the content was replaced
-        const hasToSlice = value.length - 1 === lastValue.current.length && value.slice(0, value.length - 1) === lastValue.current;
+        const hasToSlice = typeof lastValue.current === 'string' && textValue.length - 1 === lastValue.current.length && textValue.slice(0, textValue.length - 1) === lastValue.current;
 
-        // Gets the new value added by the user
-        const addedValue = hasToSlice ? value.slice(lastValue.current.length, value.length) : value;
+        // Gets the new textValue added by the user
+        const addedValue = hasToSlice && typeof lastValue.current === 'string' ? textValue.slice(lastValue.current.length, textValue.length) : textValue;
```

Comment:
maybe it can be wrapped somehow in `if(typeof lastValue.current === 'string')` instead of checking it twice

Reply:
You have any specific solution in mind? I tried applying it (as I understood it) and it made the code more complicated, maybe there is something I am not getting right 😄